src/core/expr/fexpr_count_countna.cc

```diff
@@ -0,0 +1,160 @@
+
+
+    std::string name() const override {
+      return COUNTNA? "countna"
+                    : "count";
+    }
+
+    Column evaluate1(Column&& col, const Groupby& gby, bool is_grouped) const {
+      SType stype = col.stype();
+      switch (stype) {
+        case SType::VOID:
+        case SType::BOOL:
+        case SType::INT8:
+          return make<int8_t>(std::move(col), SType::INT64, gby, is_grouped);
+        case SType::INT16:
+          return make<int16_t>(std::move(col), SType::INT64, gby, is_grouped);
+        case SType::DATE32: 
+        case SType::INT32:
+          return make<int32_t>(std::move(col), SType::INT64, gby, is_grouped);
+        case SType::TIME64: 
+        case SType::INT64:
+          return make<int64_t>(std::move(col), SType::INT64, gby, is_grouped);
+        case SType::FLOAT32:
+          return make<float>(std::move(col), SType::INT64, gby, is_grouped);
+        case SType::FLOAT64:
+          return make<double>(std::move(col), SType::INT64, gby, is_grouped);
+        case SType::STR32:
+        case SType::STR64:
+          return make<CString>(std::move(col), SType::INT64, gby, is_grouped);                
+        default:
+          throw TypeError()
+            << "Invalid column of type `" << stype << "` in " << repr();
+      }
+    }
+
+    template <typename T>
+    Column make(Column&& col, SType stype, const Groupby& gby, bool is_grouped) const {
+      if (is_grouped) {
+        return Column(new Latent_ColumnImpl(new Count_ColumnImpl<T, COUNTNA, true>(
+          std::move(col), stype, gby
+        )));
+      } else {
+        return Column(new Latent_ColumnImpl(new Count_ColumnImpl<T, COUNTNA, false>(
+          std::move(col), stype, gby
+        )));
+      }
+    }
+};
+
+
+// gets the count of all rows - nulls are not checked
+template<bool COUNTNA>
+class FExpr_CountNA_AllRows : public FExpr_Func {
+  public:
+    FExpr_CountNA_AllRows(){}
+
+    std::string repr() const override {
+      std::string out = COUNTNA ? "countna(None)" 
+                                : "count()";
+      return out;
+    }
+
+  Workframe evaluate_n(EvalContext &ctx) const override {
+    Workframe wf(ctx);
+    Groupby gby = ctx.get_groupby();
+    Column col;
+
+    if (!gby) {
+      gby = Groupby::single_group(wf.nrows());
+    } 
+
+    if (COUNTNA) {
+      col = Const_ColumnImpl::make_int_column(gby.size(), 0, SType::INT64);
+      wf.add_column(std::move(col), std::string(), Grouping::GtoONE);
+      return wf;
+    }
+    
+    if (ctx.has_groupby()) {
+      col = Column(new Latent_ColumnImpl(new CountAllRows_ColumnImpl(gby)));
+    } else {
+        auto value = static_cast<int64_t>(ctx.nrows());
+        col = Const_ColumnImpl::make_int_column(1, value, SType::INT64);
+    }
+    wf.add_column(std::move(col), "count", Grouping::GtoONE);
+    return wf;
+  }
+
+};
+
+
+static py::oobj pyfn_count(const py::XArgs &args) {
+  auto count = args[0].to_oobj_or_none();
```

Comment:
Better to call this variable `arg` or something, because `count` is the function name.